File: sdk/python/test_chunk.py
import asyncio
import os
from feedo.client import FeedoClient

async def main():
    size = 11 * 1024 * 1024
    print(f"Generating {size} bytes...")
    data = os.urandom(size)
    
    privKey = "0x0123456789abcdef0123456789abcdef0123456789abcdef0123456789abcdef"
    client = FeedoClient(private_key=privKey)
    
    print("Testing upload_bytes...")
    hash_id = await client.storage.upload_bytes(data, "test.bin")
    print(f"Uploaded! Hash: {hash_id}")

    # We won't test download immediately because DHT takes a few seconds to propagate.
    # print("Testing download_file...")
    # await client.storage.download_file(hash_id)

    print("Testing upload_private_file...")
    priv_hash = await client.upload_private_file(data, index_for_search=False)
    print(f"Uploaded private! Hash: {priv_hash}")

if __name__ == "__main__":
    asyncio.run(main())

File: sdk/typescript/test_chunk.js
const { FeedoClient } = require('./dist/index.js');
const crypto = require('crypto');
const { ethers } = require('ethers');

async function main() {
    const size = 11 * 1024 * 1024;
    console.log('Generating ' + size + ' bytes...');
    const data = crypto.randomBytes(size);

    const privKey = '0x0123456789abcdef0123456789abcdef0123456789abcdef0123456789abcdef';
    const client = new FeedoClient({ privateKey: privKey });

    console.log('Registering DID...');
    try {
        const wallet = new ethers.Wallet(privKey);
        const pubKey = wallet.signingKey.publicKey;
        await client.consensus.registerDid(pubKey, "dummy_sig");
        console.log('DID registered.');
    } catch (e) {
        console.log('DID might already be registered or error:', e.message);
    }

    console.log('Testing raw file upload (uploadFile)...');
    const t1 = Date.now();
    const hash = await client.storage.uploadFile(data, 'large_file.bin');
    console.log('Uploaded! Hash: ' + hash + ' in ' + (Date.now() - t1) + 'ms');

    console.log('Testing raw file download (downloadFile)...');
    const t2 = Date.now();
    const downloaded = await client.storage.downloadFile(hash);
    const downloadedBuf = Buffer.from(downloaded);
    console.log('Downloaded ' + downloadedBuf.byteLength + ' bytes in ' + (Date.now() - t2) + 'ms');
    
    if (downloadedBuf.equals(data)) {
        console.log('SUCCESS: Raw downloaded data matches original data!');
    } else {
        console.error('ERROR: Raw downloaded data DOES NOT match!');
    }

    console.log('\nTesting private E2EE upload...');
    const t3 = Date.now();
    const privHash = await client.uploadPrivateFile(data, undefined, false);
    console.log('Uploaded! Hash: ' + privHash + ' in ' + (Date.now() - t3) + 'ms');

    console.log('Testing private E2EE download...');
    const t4 = Date.now();
    const privDownloaded = await client.downloadPrivateFile(privHash);
    console.log('Downloaded ' + privDownloaded.byteLength + ' bytes in ' + (Date.now() - t4) + 'ms');
    
    if (privDownloaded.equals(data)) {
        console.log('SUCCESS: Private downloaded data matches original data!');
    } else {
        console.error('ERROR: Private downloaded data DOES NOT match!');
    }
}

main().catch(console.error);
